### PR TITLE
Make couchbase files modular.

### PR DESCRIPTION
**Pull request by mshima**
Couchbase is not yet supported by jhipster 7, so modularize it as a concept.

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/entity-server/files-couchbase.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const { SERVER_MAIN_SRC_DIR, SERVER_MAIN_RES_DIR } = require('../generator-constants');
+
+const entityCouchbaseFiles = {
+  dbChangelog: [
+    {
+      condition: generator => generator.searchEngineCouchbase && !generator.skipDbChangelog,
```
Comment (Tcharl) [inline generators/entity-server/files-couchbase.js:24]: SearchEngine really?

File: generators/server/files.js [hunk 2]
```diff
@@ -247,6 +153,7 @@ const cassandraFiles = {
       path: DOCKER_DIR,
       templates: [
         // docker-compose files
+        'cassandra.yml',
```
Comment (Tcharl) [inline generators/server/files.js:156]: good one :-)

**Comment by Tcharl** [comment]
Good to me

File: generators/entity-server/files-couchbase.js (hunk 1 above)
Comment (mshima) [reply generators/entity-server/files-couchbase.js:24]: Looks correct.

**Pull request merged**